arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTranslator.java

```diff
@@ -133,6 +133,15 @@ private String translateMatch2(RexNode node) {
       return translateUnary("isnull", (RexCall) node);
     case IS_NOT_NULL:
       return translateUnary("isnotnull", (RexCall) node);
+    case IS_NOT_TRUE:
+      return translateUnary("isnottrue", (RexCall) node);
+    case IS_NOT_FALSE:
+      return translateUnary("isnotfalse", (RexCall) node);
+    case INPUT_REF:
```

Comment:
I don't understand why this assumes the type of the result is boolean.

Reply:
This is because of the `TreeBuilder.makeCondition `interface feature of arrow, which requires a function to generate the corresponding filter condition, and `istrue` is such a function for it.

Reply:
why is this function translating a filter?

Reply:
I'm not sure, because when the filter structure is passed in, a node node is passed in, which forces me to translate is true

Reply:
```
    String plan = "PLAN=EnumerableCalc(expr#0=[{inputs}], booleanField=[$t0], $condition=[$t0])\n"
        + "  ArrowToEnumerableConverter\n"
        + "    ArrowProject(booleanField=[$7])\n"
        + "      ArrowTableScan(table=[[ARROW, ARROWDATATYPE]], fields=[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]])\n\n";
```
This is the untranslated plan. I hope it helps you. Since I don’t use Calcite in my work, I am not very clear about this part for now.

Reply:
I thought about it and it is OK to translate is true. Calcite will trim simple is true, such as `where 1 is true`, because they are all constants. But `booleanField is true `is obviously not.
@mihaibudiu What do you think?

Reply:
This is never a constant, it's a reference to a field within the row of the input relation, which can have any type.
Why do you think the type is boolean at this point?
Can you add an assertion that the type is always Boolean?

Reply:
The JavaDoc on this function is actually wrong, which leads to the confusion.
The JavaDoc should say that node is always an expression that evaluates to a Boolean value. This is only called from translateAnd. With this information the implementation makes sense.

Reply:
@mihaibudiu Oh, I see, thanks for reminding me. I have improved the JavaDoc